pkg/credentialprovider/provider.go

```diff
@@ -83,7 +83,7 @@ func (d *defaultDockerConfigProvider) Enabled() bool {
 // Provide implements dockerConfigProvider
 func (d *defaultDockerConfigProvider) Provide() DockerConfig {
 	// Read the standard Docker credentials from .dockercfg
-	if cfg, err := ReadDockerConfigFile(); err == nil {
+	if cfg, err := ReadDockerConfigFile(nil); err == nil {
```

Comment:
I think correctness trumps convenience... there is not a single set of paths that makes sense for both kinds of docker config files